CoEpi/repo/AlertRepo.swift

```diff
@@ -1,35 +1,64 @@
 import Foundation
 import RxCocoa
 import RxSwift
+import os.log
 
 protocol AlertRepo {
     var alerts: Observable<[Alert]> { get }
-    var updateReportsState: Observable<OperationState<CenReportUpdateResult>> { get }
+    var updateReportsState: Observable<VoidOperationState> { get }
 
     func removeAlert(alert: Alert)
     func updateReports()
 }
 
 class AlertRepoImpl: AlertRepo {
     private let cenReportsRepo: CENReportRepo
-    private let coEpiRepo: CoEpiRepo
+    private let alertsFetcher: AlertsFetcher
+    private let alertDao: RawAlertDao
 
-    lazy var updateReportsState: Observable<OperationState<CenReportUpdateResult>> = coEpiRepo.updateReportsState
+    private let updateReportsStateSubject: BehaviorRelay<VoidOperationState> = BehaviorRelay(value: .notStarted)
+    lazy var updateReportsState: Observable<VoidOperationState> = updateReportsStateSubject.asObservable()
 
     lazy private(set) var alerts: Observable<[Alert]> = cenReportsRepo.reports
         .map { reports in reports.map { $0.toAlert() }}
 
-    init(cenReportsRepo: CENReportRepo, coEpiRepo: CoEpiRepo) {
+    init(cenReportsRepo: CENReportRepo, alertsFetcher: AlertsFetcher, alertDao: RawAlertDao) {
         self.cenReportsRepo = cenReportsRepo
-        self.coEpiRepo = coEpiRepo
+        self.alertsFetcher = alertsFetcher
+        self.alertDao = alertDao
     }
 
     func removeAlert(alert: Alert) {
         cenReportsRepo.delete(report: alert.report)
     }
 
+    // TODO review thread safety with FFI/Rust: What happens if background task and foreground update
+    // TODO (e.g. pull to refresh) enter at the same time? Do we need to enqueue/lock?
```

Comment:
Background task could check what was the last update time.

Reply:
That would make sense. This is more about possible race conditions (we're also doing early exit if there's already an operation in progress).